### DISCUSSION TITLE
About model persistence

### DISCUSSION THREAD

**Discussion by gaosiy** [intel/scikit-learn-intelex#1212] [Q&A] [answered]
Hi everyone, I use intelex to speed up my svm training. But when I try to save the model and use it on another device, it report a error "ModuleNotFoundError: No module named 'sklearnex' ".  I think the intelex may add something to the model which I do not need. because the new device is an arm-linux device, I can not install sklearnex on it.

 My demo code show here, first, I use svm without intelex and save the model.
`from sklearn.svm import SVC
from sklearn import metrics
from sklearn.datasets import fetch_openml
from sklearn.model_selection import train_test_split

x, y = fetch_openml(name='a9a', return_X_y=True)
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)
params = {
    'C': 100.0,
    'kernel': 'rbf',
    'gamma': 'scale'
}

classifier = SVC(**params).fit(x_train, y_train)

from joblib import dump, load
dump(classifier, './model/svm_normal.joblib') `

Then I used intelex and saved the model
`from sklearnex import patch_sklearn
patch_sklearn()
params = {
    'C': 100.0,
    'kernel': 'rbf',
    'gamma': 'scale'
}
start = timer()
classifier_intel = SVC(**params).fit(x_train, y_train)
from joblib import dump, load
dump(classifier_intel, './model/svm_sklearnex.joblib') `

I check the tow models, they are diferent.
![image](https://user-images.githubusercontent.com/16640004/224982283-f2be275c-22f7-498a-9d9c-5c74e94bd954.png)

How can I use speed up in the training step, and use as a normal sklearn model in inference step?
Thanks and best wishes!

Yan

**Comment by napetrov** [comment, marked as answer]
@gaosiy With current design there is no support general support for non-sklearnex inference. This is area that we would consider looking  on, but not in nearest future.

Also it's important to note that skelearnex not necessary creates scikit-learn compatible models and this is part of optimizations that are used.  So simple linear models 1:1 mapping is possible, while for more complex models such as SVM, Random Forest models would be different.
